### DISCUSSION TITLE
device.interfaces.all() does not show all interfaces of virtual chassis

### DISCUSSION THREAD

**Discussion by peterbaumert** [netbox-community/netbox#12475] [Help Wanted!] [answered]
Hi all
i am trying the render config feature to render my switch configs.

When i do device.interfaces.all() it gives me only the interfaces of this device itself not all of the virtual chassis. This is the primary device which in the web interface shows all interfaces.

No sure if this is a bug or i have to do this different, any ideas?

KR
Pete

**Comment by candlerb** [comment]
Firstly, to be clear: is this the Django ORM you are using, or pynetbox with the REST API?

Secondly: is the "device" that you are querying the master device in the virtual chassis? (Sounds like yes, since you say viewing this in the UI shows all interfaces)

**Comment by peterbaumert** [comment]
Uhm i would say its the Django ORM. I am using the new render config feature and use this inside of the jinja2 template.

Second question, yes its the master :)

**Reply by candlerb** [reply]
Thanks. You can exercise the ORM interactively using [nbshell](https://docs.netbox.dev/en/stable/administration/netbox-shell/). Do you see the same behaviour there?

**Reply by DanSheps** [reply, marked as answer]
Device.vc_interfaces returns an object manager for all VC members I believe, whereas interfaces is just for the "local" device.

**Reply by peterbaumert** [reply]
Thank you very much, didn't find that one. It works perfectly now!
